fw/tls.c

```diff
@@ -170,7 +170,7 @@ tfw_tls_connection_recv(TfwConn *conn, struct sk_buff *skb)
 
 		/* Do upcall to http or websocket */
 		r = tfw_connection_recv(conn, data_up.skb);
-		if (r == TFW_BLOCK) {
+		if (r == TFW_BLOCK || r == TFW_BAD) {
 			kfree_skb(nskb);
```

Comment:
I'm wondering if there are other similar places. It'd be good to run the testsuite with some memory leakage detector, like kmemleak. However, it doesn't track buddy allocator while we use it for skbs. IIRC @s0nx recently shared a link to a kernel debugging tool, which catches page leaks, but I can't find a link...

It'd be good to add a chapter about memory leak detection to https://github.com/tempesta-tech/tempesta/wiki/Debugging-and-troubleshooting with the link and probably link to kmemleak

Reply:
I used this tool to track page allocations: [page_owner](https://www.kernel.org/doc/html/v5.0/vm/page_owner.html)

Reply:
@EvgeniiMekhanik did you have a chance to test the code for memory leaks? @pale-emperor @RomanBelozerov do we have a memory leakage and/or stress tests for HTTP/2?

Reply:
I try to use page_owner, but i can't find some useful information in it's logs. I will ask about it @s0nx on our meeting